src/input_ir.rs

```diff
@@ -1,9 +1,35 @@
 //! Input data structures to the audio graph compiler.
-//!
+
+#[cfg(feature = "serialize")]
 use serde::{Deserialize, Serialize};
 
-/// The input IR used by the audio graph compiler.  
-#[derive(Clone, Debug, Serialize, Deserialize)]
+/// A globally unique identifier for a [Node].
+#[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct NodeID(pub u64);
+
+/// The ID for a [Port] on a particular [Node].
+///
+/// This does not need to be a globally unique identifier,
+/// just unique to the [Node] it belongs to.
+#[cfg_attr(feature = "serialize", derive(Serialize, Deserialize))]
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct PortID(pub u32);
```

Comment:
Why `u32` here instead of `u64` like the rest?

Reply:
I suppose this could be a `u64`.

I made it a `u32` because it directly correlates with how clap identifies its ports: <https://github.com/free-audio/clap/blob/main/include/clap/ext/audio-ports.h#L45>

I just figured that `u64` was overkill because each port ID no longer needs to be unique to the entire graph, just unique to the node it belongs to.

Reply:
Oh, I just saw the note that clap's port IDs may now overlap between inputs and outputs. I'll fix that so it's consistent here too.

Reply:
> I just figured that `u64` was overkill because each port ID no longer needs to be unique to the entire graph, just unique to the node it belongs to.

Because we are using these IDs as keys in a hashmap, it's actually more efficient to use `u32` over `u64`. In fact, I would rather have all IDs in the graph use `u32` instead.

Reply:
I'm ok with switching everything to u32